helios-client/src/main/java/com/spotify/helios/client/HeliosClient.java

```diff
@@ -511,6 +511,7 @@ public static Builder newBuilder() {
     private ClientCertificatePath clientCertificatePath;
     private Supplier<List<Endpoint>> endpointSupplier;
     private boolean sslHostnameVerification = true;
+    private ListeningScheduledExecutorService executorService = defaultExecutorService();
```

Comment:
I think this should be set in the `build()` method instead so that an executor is not created/started here and then replaced for people calling `setExecutorService`